react/frontpage/components/ShowsGraph.jsx

```diff
@@ -0,0 +1,215 @@
+// ShowsGraph.jsx
+
+var React = require('react');
+
+import Dates from '../../common/Dates';
+
+// styling
+require('./shows.scss');
+
+const week = ["sunday", "monday", "tuesday", "wednesday", "thursday", "friday", "saturday"];
+const dayWidth = `${100/8}%`;
+
+/*
+Full graph with schedule of shows
+
+@prop shows: list of show objects
+
+@prop currentShowID: id of current show
+@prop spotlightShowID: id of spotlight show
+@prop activeShowID: id of active show user clicked on
+@prop onShowClick(show): callback for when a show is made active
+*/
+const ShowsGraph = React.createClass({
+	getInitialState: function() {
+		return {};
+	},
+	componentWillMount() {
+		this.updateSchedule(this.props.shows);
+	},
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.shows) {
+			this.updateSchedule(nextProps.shows);
+		}
+	},
+
+	updateSchedule: function(shows) {
+		var sorted = sortedShows(shows);
+		this.setState({
+			schedule: sorted
+		});
+	},
+	findStartEndTimes: function(start, end) {
+
+		for(start=0; start < 24; start ++){
```

Comment:
Also, if this variable is indexing `Dates.availableTimes`, can we iterate to `< Dates.availableTimes` Same below